src/test/java/seedu/task/logic/EditCommandTest.java

```diff
@@ -6,32 +6,24 @@
 import seedu.task.logic.commands.EditTaskCommand;
 import seedu.task.model.TaskBook;
 import seedu.task.model.item.Task;
+import seedu.taskcommons.core.Messages;
 
 public class EditCommandTest extends CommandTest {
 
-    /* Need help with this test to test for duplicates not allowed. By running the program, the exception can be caught
-     * */
-
-    @Ignore
     @Test
     public void execute_editFloatTask_duplicate() throws Exception {
         // setup expectations
         TestDataHelper helper = new TestDataHelper();
         Task toBeAdded = helper.computingFloatTask();
         TaskBook expectedAB = new TaskBook();
         expectedAB.addTask(toBeAdded);
-        toBeAdded = helper.computingEditedNameFloatTask();
-        expectedAB.addTask(toBeAdded);
-        
-        // setup model
-        model.addTask(helper.computingFloatTask());
-        model.addTask(helper.computingEditedNameFloatTask());
+        Task toBeAdded2 = helper.computingEditedNameFloatTask();
+        expectedAB.addTask(toBeAdded2);
         Task toBeEdited = helper.computingFloatTask();
-        model.editTask(toBeEdited, 0);
 
         // execute command and verify result
-        assertEditTaskCommandBehavior(helper.generateAddTaskCommand(toBeAdded),helper.generateListTaskCommand(),
-                helper.generateEditTaskCommand(toBeEdited),
+        assertEditTaskCommandBehavior(helper.generateAddFloatTaskCommand(toBeAdded), helper.generateAddFloatTaskCommand(toBeAdded2),helper.generateListTaskCommand(),
+                helper.generateEditTaskCommand(toBeEdited,1),
                 String.format(EditTaskCommand.MESSAGE_DUPLICATE_TASK, toBeEdited),
                 expectedAB,
                 expectedAB.getTaskList());
```

Comment:
hmmm ok I'm not so sure of number 1, since I added my tasks into the model directly after added them in expectedAB. 

Hmmm ok I understand, so you are not testing the editing of task but the editing of a duplicate. Just a thought though, when you mean duplicate, does it mean editing it to the same thing or editing it to have a same name as an existing task in the list?